### ISSUE TITLE
English Forum - or Forum category on this GitHib

### ISSUE THREAD

**Issue by nick0** [evolution-cms/evolution#1828]
Hello Amazing EvolutionCMS Devs,

Can we please have a forum. In English.

Or a Forum Category on GitHib.

The Telegram chat is a chat. It is not appropriate for detailed CMS related queries and responses. It's very hard to find important information and replies.

A chat based system where questions are firstly automatically fed responses from a robot is not the way to grow the Evolution user base. The default human reply is often read the docs. This can be frustrating for non technical users when there are, what, 4 different document sites and not one covers every scenario, nor has comprehensive examples. Lots of reading between the lines or assumed knowledge is required which is hard for non technical users.

Please give us something that we can search and bookmark and easily access the same info again and ask questions without being referred to the docs each time. Like we used to have on the old Modx Evolution Forums - that was so handy.

Don't you want to increase the user base of Evolution beyond existing, die hard Modx Evolution users? 

Surely a forum in English would help this?

This is not in any way a criticism. It is a plea for a forum that does not exclude English users.

Please.

**Comment by ArnoNymeier** [comment]
I wish that too!
In addition, it would be very helpful if more detailed information and, if possible, a "how to do" were given to the packages in the package manager. Here, too, a lot is not in English and at least for me sometimes not understandable from a technical point of view.
I think for non-technical users, examples are the best way to understand functions and integrations step by step.

**Comment by Dmi3yy** [comment]
In plans do new site with forum (multilangs (google auto-translate)),  do forum for Engish users  only not help with growth community because lot information only in russian :( 

unfortunately the resources of the active part of the community are not enough for everything, so some of the important things that need to be done are postponed. Prior to release, the main focus was on development. Now we are planning a video course and the creation of a new demo site.

**Comment by fourroses666** [comment]
I can see allot of benefits with a forum indeed. Back in the day, long time ago I always liked and used the forum allot.

**Comment by modxuser** [comment]
+1 on this

I am currently converting a site to Doclister and have no problem creating examples / tutorials for what I am doing, but, there's nowhere to add them.

As I re-wrote the Evo docs I was able to add tutorials there, but I no-longer have access to it and to be honest, the content is now way out of date. The formlister tutorials that I added are still up-to-date, but the rest is pretty much useless.

If snippet questions are answered, then they to can be added to tutorials, that way we non-developers don't have to keep getting on developers nerves asking one and the same question time after time

**Comment by Dmi3yy** [comment]
We have nice docs based on github but, not see any help here with translate or add new information. 

https://docs.evo.im/en/

**Comment by Dmi3yy** [comment]
if no one helps with the documentation, how will the forum help?

**Comment by nick0** [comment]
As a place to discuss / ask questions etc. In English. Like the responses to some of these recent posts.

**Comment by fourroses666** [comment]
Yes, we need to translate the official docs. https://docs.evo.im/en/ 
Which can be found here: https://github.com/evolution-cms/docs/
(everybody has access to it when having an github account, just click on the pencil icon to start translating)

But also I would like to have a platform/form where we can discuss issues in English.
Not only translate Russian discussions :(

**Comment by Dmi3yy** [comment]
No problem i can install any ready solution for that

**Comment by modxuser** [comment]
I don't wish to tread on toes, we have had this discussion too many times and nothing has come of it.

I spent 7 hours last night looking through the Russian forum using Google translate, the translation was a mess to say the least. For a non-russian speaker trying to translate Russian is a nightmare, that's why I gave up on it over a year ago.

The Russian community is the largest using Evo, that's fact, as such, they seem to take priority and all others have to see how they get along, which lets be honest, we do, but not very well.

I haven't personaly been active here for such a long time simply because a lot has been said in the past, but it has never happend.

I am currently building a new site, which I was going to build using Revo, simply because there is more support due the English community, but I decided to go with Evo, so I hope it wasn't an error to do so.

Fact is, the English community needs to be more active, fact is, the English community needs a place to ask and answer questions, fact is the Evo Dev team have done a great job and if Evo is to become larger a place for International users needs to be built.

If my post offends anyone, sorry, but this is my opinion

**Comment by modxuser** [comment]
If a ready solution can be installed, please do so, that would be great

If a section for tips, tricks and tutorials can be created, even better

**Comment by Dmi3yy** [comment]
Plz try this: https://evo.flarum.cloud
if all ok i install this like forum.evo.im )

**Comment by risingisland** [comment]
I feel that a big portion of the English community simply left to find an alternative solution.
For most, having to jump through hoops to find info, even if it is using Google Translate, just isnt comfortable, and adds too much back and forth. Those extra steps add up, time wise.

**Comment by Dmi3yy** [comment]
done: 
https://forum.evo.im   (SSL will work from tomorrow)

**Comment by Dmi3yy** [comment]
Now the English-speaking community is very small, and also uses things that have long been outdated.

The core team is mainly engaged in the development of the kernel, but no one is involved in promoting to the masses :(

I want to develop the system for the English-speaking community, but I need help. For example, everyone wants documentation in English, but no one is ready to translate

**Comment by modxuser** [comment]
Translating Russian into English is a nightmare for a non-Russian speaker, I've tried, the result is not understandable

I think the English speaking forum is a start, I think we are all moving towards the new snippets and as such we need to build tutorials - I will add some tutorials as soon as the forum is up and running

The docs need to have all of the rubbish removed, including all of the old outdated snippets that have long gone, but some of the standard stuff needs to be kept for the time being e.g Ditto, Wayfinder

**Comment by Dmi3yy** [comment]
I think it the same like Japan to Russian :) i spend many time for that when Japan EVO grew faster then main branch :) 

Yep it's hard but need start, if have question i help, i need more practice in English )  You can write me directly with question

**Comment by Dmi3yy** [comment]
Now work with SSL 
https://forum.evo.im

Ready for use ) and who want be moderators write plz here

**Comment by bossloper** [comment]
Is it sending setup confirmation emails correctly?

**Comment by fourroses666** [comment]
Could be in spam bit I did got it

**Comment by modxuser** [comment]
moderator +1

I am happy to give it a go so long as I can ask for help if a problem arises :)

**Comment by bossloper** [comment]
Strange. If I use a Google email account I get the confirmation. But if I use a Microsoft account I do not get the confirmation email.

@Dmi3yy - I am also ok with being a moderator (MrB)

**Comment by fourroses666** [comment]
Moderator is oke for me too. Lets see how this forum works out :)

**Comment by ArnoNymeier** [comment]
That's great news. Registration in the new forum worked without any problems. :-)

**Comment by fourroses666** [comment]
https://discuss.flarum.org/d/18491-imgur-upload-simple-image-upload-from-the-editor/30

I would like to see this.

**Comment by nick0** [comment]
Thanks so much @Dmi3yy and all in this thread. This is amazing news! So awesome :)

I am still waiting on the confirmation email to come (hours later) (not in junk... nowhere)

**Comment by bossloper** [comment]
@nick0 - Sounds like you are having the same signup issues I had. There is definitely something not right on the https://forum.evo.im . But the test setup worked fine for me: https://evo.flarum.cloud/

**Comment by Dmi3yy** [comment]
Maybe problem in mail server i aprove by database

**Comment by risingisland** [comment]
Mine was sent through immediately and approved.

**Comment by Dmi3yy** [comment]
aprove bossloper and nic0

**Comment by fourroses666** [comment]
I was thinking about:
`
Announcements

General EVO 1 & 2
General EVO 3
Random
Development
Extras
Tips & Tricks
Documentation
Jobs`

**Comment by modxuser** [comment]
@fourroses666 

I think we should walk before we run :)

Adding too much too soon, probably won't help - the community is small

I personaly think a tutorials section is an absolute must, we all know of the problems trying to translate Russian into English, so if we have tutorials / ready solutions it makes life easier understanding how to modify and add / take away bits from the code.

I propose the following initial "categories" for the Evo Forum:

1) Ask a Question - similar your General Evo 1 + 2 + 3
2) Tips & Tricks
3) Tutorials (with subsections as follows)
.... Doclister
.... DLBeforeAfter
.... DLCrumbs
.... DLFirstChar
.... DLGlossary
.... DLMenu
.... DLPrevNext
.... DLReflect
.... DLReflectFilter
.... DLSitemap
.... DLTemplate
.... DLValuelist
.... Formlister
.... iF

**Comment by risingisland** [comment]
@modxuser I think I have a bit more that could be added to your tutorial too.

**Comment by fourroses666** [comment]
These tuts would be very handy indeed. I do think we must separate v1+2 with v3. Further on I'm oke with your suggestion.
As moderator I don't think we can add  Categories. We have to ask Dmytro I guess.

**Comment by modxuser** [comment]
@risingisland 

That tutorial example can be broken down to a very simple call, a call with pagination, a call for displaying content from posts etc.

It's basically a starting point that can be built upon without any problems, I will probably modify it a little anyway, so as to incorporate all of the pagination templates - that way, whoever wants to use the code can change / remove as needed.

@fourroses666 

I think the way forward is to keep with the code that is suported by the DEV team, meaning, all snippets that they class as "old" should have tutorials for their new equivalents, Ditto => Doclister, Wayfinder => DLMenu, Crumbs => DLCrumbs etc.

I can supply some DLMenu tutorials, one for DLCrumbs, a few for Doclister (although for Doclister you only realy need one and that then gets modified to fit the situation) and I have already created numerous for FormLister when I re-built the Evo Docs site (which is still online) - I have a copy here anyway, so they can be added to the Forum as well.

All other snippets that are not seen as worthy by the DEV team and are still used intensively need to be reviewed to see if there isn't already a different solution.

Custom snippets / code is not a problem that should be IMHO added to the forum, as it's custom.

All I need to get started / add the tutorials are the categories in the Forum - @Dmi3yy

**Comment by modxuser** [comment]
@fourroses666 , @bossloper , @nick0 , @risingisland , @ArnoNymeier, @Dmi3yy 

May I suggest the following "initial" categories for the forum:

```
1) General EVO 1 & 2
2) General EVO 3
3) Tips & Tricks
4) Snippet Tutorials (with subsections as follows)
.... Doclister
.... DLBeforeAfter
.... DLCrumbs
.... DLFirstChar
.... DLGlossary
.... DLMenu
.... DLPrevNext
.... DLReflect
.... DLReflectFilter
.... DLSitemap
.... DLTemplate
.... DLValuelist
.... Formlister
.... iF
```

If anyone is not happy, please say here :)

**Comment by fourroses666** [comment]
`Snippet Tutorials` should be `Extras tutorials` or just `Tutorials`.

That are allot of Extras by the way, maybe start with: (and add others "on demand"?)
.... DocLister
.... DLMenu
.... DLSitemap
.... FormLister
.... iF

**Comment by modxuser** [comment]
All of the "DLxxx"  snippets are part of Doclister and as we saw the other day, they work using Doclister, but they are called as single snippets

**Comment by modxuser** [comment]
I am currently using and have converted to:

```
.... Doclister
.... DLBeforeAfter
.... DLCrumbs
.... DLGlossary
.... DLMenu
.... DLPrevNext
.... DLSitemap
.... Formlister
.... iF
```

**Comment by Dmi3yy** [comment]
Category it tags, 

i test today or add admin access

**Comment by Dmi3yy** [comment]
Up rights to admin level for 
modxuser, fourroses666, bossloper

**Comment by bossloper** [comment]
Guess we need to have a play with it. I see there is a 'restrict by tag' option. Not sure how that works yet, but maybe we can create an 'Admins' tag, restrict it, and start a thread with how we are going to set it all up?

**Comment by fourroses666** [comment]
I have added some primary Tags. I guess DocLister and others need a secondary Tag.

**Comment by fourroses666** [comment]
Yes, we can discuss further on the Forum, I see there is an Admin Tag added and we discuss there.

**Comment by bossloper** [comment]
@Dmi3yy  - Signup emails are working fine now. Whatever you did - thanks.

**Comment by fourroses666** [comment]
SSL isn’t working yet thow.

**Comment by risingisland** [comment]
> 
> 
> SSL isn’t working yet thow.

Its working for me.

**Comment by bossloper** [comment]
and me.

**Comment by fourroses666** [comment]
Hmmm indeed, Firefox is still bitching but I think its somekind of browsercache..

**Comment by modxuser** [comment]
@fourroses666 

What I recently did after an update due to FireFox resetting stuff:

type in the address bar **about:config**

then press the button i'l be careful i promise

then type in the bar **browser.cache.disk.enable**

then double click on it, to make it false

exit firefox and restart-it.

do the same with **browser.cache.memory.enable**

if you want to read more:

http://kb.mozillazine.org/Browser.cache.disk.enable

http://kb.mozillazine.org/Browser.cache.memory.enable

**Comment by fourroses666** [comment]
We have a forum now: https://forum.evo.im/

Can be closed.

**Comment by Dmi3yy** [comment]
Add Discussions on github to 
https://github.com/evolution-cms/evolution/discussions

**Comment by modxuser** [comment]
@Dmi3yy 

> Add Discussions on github to
> https://github.com/evolution-cms/evolution/discussions

Is this supposed to be a replacement for the brand new forum ?

**Comment by Dmi3yy** [comment]
i remove wiki, think discussions better then wiki. 

and want all info issues move to discussions 

and think we can use both forum and discussions

**Comment by modxuser** [comment]
@Dmi3yy 

maybe calling it "issues" would be better than "discussions" as discussions has a different meaning altogether

**Comment by Dmi3yy** [comment]
this names from github 

for me: 
- issues bugs and task what need do 
- discussions for Q&A and another info or question were no need any task with fix code

**Comment by modxuser** [comment]
The problem I personaly see is spreading information between too many places

Discussions should be in the forum, be it Q&A or other code questions that don't need code being fixed

The new forum already has multiple tutorials, discussions and answers

**Comment by fourroses666** [comment]
I think Dmi3yy would like to keep the bugs&fixes (issues) clean and ship the questions which might not be bugs to another part of Github. There for "Discussions" can be used but I don't think it suppose to replace the forums EN or RU.

**Comment by modxuser** [comment]
@fourroses666 

I fully understand what @Dmi3yy  wrote, his English is fully understandable and clear

What I am trying to say is adding  a place in GitHub to ask questions about "how to do this" or "how to do that" is in my opinion the wrong place.

I was always under the impression that the DEVs wanted such questions out of GitHub, that's the whole idea of moving these questions and answers to forums

**Comment by bossloper** [comment]
Keeping bugs&fixes clean makes good sense and is overdue :-).

Any thread of conversation be it forum or on GitHub discussion may or may not end up in a code change 'fix' or 'enhancement'. At which point they would presumably be asked to open a GitHub issue... that is how I remember it working back in the old MODx forum. 
There seems a big overlap between code discussion on GitHub and forum discussion. Knowing if to post in GitHub discussion or a forum will confuse people and over time one will win out for one reason or another, be it convenience/usability etc.

**Comment by Dmi3yy** [comment]
Yep but who will moderate when some user write in wrong place? 

I just want do clear issues, its help see what need do 

now we have 242 issues but i think for work ready only 50, all another old or only info

**Comment by modxuser** [comment]
The forum is setup with moderators and admins, as you know, you set it up :)

But moderators or admins for discussions in GitHub is beyond me

**Comment by risingisland** [comment]
Not that my opinion matters, but I too am wondering how many platforms we need to spread this across.
We now have:
• GitHub Issues
• Telegram
• Slack
• English Forum
• GitHub Discussions
• Evo Ru Community
• etc.

**Comment by Dmi3yy** [comment]
the bigger, the better.

so it will be easier to find information about the Evo.

**Comment by Dmi3yy** [comment]
Now most usage: 
-telegram
- github 
- forum.evo.im

**Comment by nick0** [comment]
> this names from github
> for me:
>     * issues bugs and task what need do
>     * discussions for Q&A and another info or question were no need any task with fix code

If Discussions is misleading for some, perhaps it could be called Questions or Q&A?

> the bigger, the better.
> so it will be easier to find information about the Evo.

I think less is more. The more places, the more diluted and harder to find info.

**Comment by modxuser** [comment]
> The more places, the more diluted and harder to find info

I fully agree on that
